components/hab/src/license.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{common::ui::{self,
+                         UIReader,
+                         UIWriter,
+                         UI},
+            error::{Error,
+                    Result},
+            hcore::{fs::{am_i_root,
+                         FS_ROOT_PATH},
+                    users::get_current_username}};
+use chrono::prelude::*;
+use dirs;
+use serde_yaml;
+use std::{fs::{self,
+               File},
+          io::Write,
+          path::PathBuf};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LicenseData {
+    pub date_accepted: DateTime<Utc>,
+    pub accepting_product: String,
+    pub accepting_product_version: String,
+    pub user: Option<String>,
+    pub file_format: String,
+}
+
+impl LicenseData {
+    pub fn new() -> Self {
+        LicenseData { date_accepted: Utc::now(),
+                      accepting_product: String::from("hab"),
+                      accepting_product_version: super::VERSION.to_string(),
+                      user: get_current_username(),
```

Comment:
Should this actually be a real person's name? As is, I think we're going to have a lot of licenses accepted by `root`.

Reply:
It's fine.  The metadata isn't real important.  Users can preemptively touch a blank file in the right spot and we will consider that license acceptance.